runtime/src/utils/buffer/write.rs

```diff
@@ -150,10 +167,10 @@ impl<B: Blob> Write<B> {
         assert!(capacity > 0, "buffer capacity must be greater than zero");
 
         Self {
-            inner: Arc::new(RwLock::new(Inner {
-                blob,
-                buffer: Vec::with_capacity(capacity),
-                position: size,
+            blob,
+            buffer: Arc::new(RwLock::new(Buffer {
```

Comment:
nit: I think this probably makes sense to create a Buffer constructor

Reply:
done